Corona-Warn-App/src/main/java/de/rki/coronawarnapp/srs/ui/typeselection/SrsTypeSelectionFragment.kt

```diff
@@ -1,17 +1,108 @@
+
     private val binding by viewBinding<FragmentSrsTypeSelectionBinding>()
 
+    @Inject lateinit var itemAdapter: SrsTypeSelectionItemAdapter
+
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
+
+        viewModel.navigation.observe2(this) {
+            when (it) {
+                SrsTypeSelectionNavigationEvents.NavigateToCloseDialog -> showCancelDialog()
+                SrsTypeSelectionNavigationEvents.NavigateToMainScreen ->
+                    findNavController().navigate(
+                        SrsTypeSelectionFragmentDirections.actionSrsSubmissionTypeSelectionFragmentToMainFragment()
+                    )
+                SrsTypeSelectionNavigationEvents.NavigateToShareCheckins ->
+                    findNavController().navigate(
+                        SrsTypeSelectionFragmentDirections
+                            .actionSrsSubmissionTypeSelectionFragmentToSrsCheckinsFragment()
+                    )
+                SrsTypeSelectionNavigationEvents.NavigateToShareSymptoms ->
+                    findNavController().navigate(
+                        SrsTypeSelectionFragmentDirections
+                            .actionSrsSubmissionTypeSelectionFragmentToSrsSymptomsFragment()
+                    )
+            }
+        }
+
+        val backCallback = object : OnBackPressedCallback(true) {
+            override fun handleOnBackPressed() = viewModel.onCancel()
+        }
+        requireActivity().onBackPressedDispatcher.addCallback(viewLifecycleOwner, backCallback)
+
+        binding.typeList.apply {
+            layoutManager = LinearLayoutManager(requireContext())
+            adapter = itemAdapter
+        }
+
+        itemAdapter.data = typeList
+
+        itemAdapter.onItemClickListener = {
+            viewModel.selectTypeListItem(it)
+        }
+
+        binding.toolbar.setNavigationOnClickListener { viewModel.onCancel() }
+
+        viewModel.srsTestType.observe2(this) {
+            updateButtons(it)
+        }
+    }
+
+    private fun updateButtons(typeSelectionItem: SrsTypeSelectionItem) {
+
+        binding.typeList.children.forEachIndexed { i, view ->
+            (view as Button).apply {
+                handleColors(typeSelectionItem.submissionType, typeList[i].submissionType)
+            }
+        }
+
+        binding.typeSelectionNextButton.apply {
+            isEnabled = typeSelectionItem.submissionType != null
+            setOnClickListener { viewModel.onNextClicked() }
+        }
+    }
+
+    private fun Button.handleColors(submissionType: SrsSubmissionType?, state: SrsSubmissionType?) {
+        setTextColor(formatSrsTypeSelectionButtonTextStyleByState(context, submissionType, state))
+        backgroundTintList =
+            ColorStateList.valueOf(formatSrsTypeSelectionBackgroundButtonStyleByState(context, submissionType, state))
+    }
+
+    private fun showCancelDialog() {
```

Comment:
BTW  this the common close dialog https://www.figma.com/file/KuDEcMSREg3VhZRhC3FXTY/COVID19_Master_2.25?node-id=57680%3A41751
Please make it reusable for other screens and one more task is basically done ;)

Reply:
I've already did this for my symptoms intro task. @schauersbergern, you can skip it and I'll add it in my PR.

Reply:
Added CommonCloseDialog.kt

Reply:
> I've already did this for my symptoms intro task. @schauersbergern, you can skip it and I'll add it in my PR.

Sorry seen that too late, removed CommonCloseDialog.kt 😄